File: pr1-java/src/main/java/userInterfacing/UserInterface.java
package userInterfacing;

import domain.User;
import services.Service;
import utils.observers.Observer;

public abstract class UserInterface implements Observer {
    protected Service service = null;
    protected User loggedInUser = null;

    public void init(Service service, User loggedInUser) {
        this.service = service;
        this.loggedInUser = loggedInUser;
    }

    @Override
    public void update() {

    }
}

File: pr1-cs/UserInterfacing/UserInterface.cs
using System.Configuration;
using Gtk;
using pr1_cs.Domain;
using pr1_cs.Services;

namespace pr1_cs.UserInterfacing
{
    public abstract class UserInterface
    {
        protected readonly Service Service;
        protected User LoggedUser;
        protected Window OwnedWindow;

        protected UserInterface(Service service, User loggedUser)
        {
            GtkClient.OpenWindows++;
            Service = service;
            LoggedUser = loggedUser;
        }

        public void Open()
        {
            OwnedWindow.Show();
        }

        protected void Close()
        {
            OwnedWindow.Destroy();
        }
    }
}